homeassistant/components/axion_dmx/light.py

```diff
@@ -0,0 +1,290 @@
+                        scaled_rgbww = [
+                            scale_brightness(c, self._attr_brightness, 255)
+                            for c in rgbww
+                        ]
+                    await self.api.set_rgbww(self._channel, scaled_rgbww)
+        elif self._light_type == "Tunable White":
+            if ATTR_COLOR_TEMP in kwargs:
+                self._attr_color_mode = ColorMode.COLOR_TEMP
+                self._color_temp = color_util.color_temperature_mired_to_kelvin(
+                    kwargs[ATTR_COLOR_TEMP]
+                )
+            else:
+                self._color_temp = self._last_color_temp
+
+            warm_white_level, cold_white_level = get_tunable_white_levels(
+                self._color_temp,
+                warm_white_k=1800,  # Fixed value for warm white LED
+                cool_white_k=6000,  # Fixed value for cold white LED
+                max_level=255,
+                brightness=self._attr_brightness
+                if self._attr_brightness is not None
+                else 255,
+            )
+            self._last_color_temp = self._color_temp
+            _LOGGER.debug(f"Setting Warm Light level - {warm_white_level}")
+            await self.api.set_level(self._channel, warm_white_level)
+            _LOGGER.debug(f"Setting Cold White level - {cold_white_level}")
+            await self.api.set_level(self._channel + 1, cold_white_level)
+        else:
+            await self.api.set_level(self._channel, self._attr_brightness)
+
+        # Add a small delay to allow the controller to process the command
+        await asyncio.sleep(0.5)
+
+        await self.coordinator.async_request_refresh()
+        self.async_write_ha_state()
+
+    async def async_turn_off(self, **kwargs: Any) -> None:
+        """Instruct the light to turn off."""
+        _LOGGER.debug(f"Instructing the {self._name} to turn off!")
+        self._attr_is_on = False
+        await self.api.set_level(self._channel, 0)
+
+        if self._light_type in ["RGB", "RGBW", "RGBWW"]:
+            self._last_hs_color = self._attr_hs_color
+            await self.api.set_level((self._channel + 1), 0)
+            await self.api.set_level((self._channel + 2), 0)
+            if self._light_type in ["RGBW", "RGBWW"]:
+                self._last_rgbw = self._attr_rgbw_color
+                await self.api.set_level((self._channel + 3), 0)
+                if self._light_type == "RGBWW":
+                    self._last_rgbww = self._attr_rgbww_color
+                    await self.api.set_level((self._channel + 4), 0)
+        elif self._light_type == "Tunable White":
+            await self.api.set_level(self._channel + 1, 0)
+
+        # Add a small delay to allow the controller to process the command
+        await asyncio.sleep(0.5)
+
+        # Manually refresh the coordinator to get the latest state
+        await self.coordinator.async_request_refresh()
+        self.async_write_ha_state()
```

Comment:
So we want to have all the device specific code to exist in the library. I know that this code is also responsible for displaying the state of the entity correctly, but I also think that a lot of logic could be moved to the library, what do you think?

Reply:
As the lib and the device don't know anything about the light_type, I'm not sure that's reasonable enough to move all of this logic to the lib. What do you think about that?

Reply:
Hmm, it's a difficult one because the function does a lot, both with the library as with the entity. I think we could start off with removing the `await self.api.set_level((self._channel + 1), 0)`, I think the logic about what value to send to what channel is device (or protocol for that matter) specific and should belong in the library

Reply:
I just pushed new change, where I'm using already implemented library functions to turn off the lights.